presto-main/src/main/java/com/facebook/presto/sql/rewrite/DescribeInputRewrite.java

```diff
@@ -0,0 +1,145 @@
+import static com.facebook.presto.sql.QueryUtil.ordering;
+import static com.facebook.presto.sql.QueryUtil.row;
+import static com.facebook.presto.sql.QueryUtil.selectList;
+import static com.facebook.presto.sql.QueryUtil.simpleQuery;
+import static com.facebook.presto.sql.QueryUtil.values;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static java.util.Objects.requireNonNull;
+
+final class DescribeInputRewrite
+        implements StatementRewrite.Rewrite
+{
+    @Override
+    public Statement rewrite(
+            Session session,
+            Metadata metadata,
+            SqlParser parser,
+            Optional<QueryExplainer> queryExplainer,
+            Statement node,
+            List<Expression> parameters,
+            AccessControl accessControl,
+            boolean experimentalSyntaxEnabled)
+    {
+        return (Statement) new Visitor(session, parser, metadata, queryExplainer, parameters, accessControl, experimentalSyntaxEnabled).process(node, null);
+    }
+
+    private static final class Visitor
+            extends AstVisitor<Node, Void>
+    {
+        private final Session session;
+        private final SqlParser parser;
+        private final Metadata metadata;
+        private final Optional<QueryExplainer> queryExplainer;
+        private final List<Expression> parameters;
+        private final AccessControl accessControl;
+        private final boolean experimentalSyntaxEnabled;
+
+        public Visitor(
+                Session session,
+                SqlParser parser,
+                Metadata metadata,
+                Optional<QueryExplainer> queryExplainer,
+                List<Expression> parameters,
+                AccessControl accessControl,
+                boolean experimentalSyntaxEnabled)
+        {
+            this.session = requireNonNull(session, "session is null");
+            this.parser = parser;
+            this.metadata = metadata;
+            this.queryExplainer = queryExplainer;
+            this.accessControl = accessControl;
+            this.parameters = parameters;
+            this.experimentalSyntaxEnabled = experimentalSyntaxEnabled;
+        }
+
+        @Override
+        protected Node visitDescribeInput(DescribeInput node, Void context)
+                throws SemanticException
+        {
+            String sqlString = session.getPreparedStatement(node.getName());
+            Statement statement = parser.createStatement(sqlString);
+
+            // create  analysis for the query we are describing.
+            Analyzer analyzer = new Analyzer(session, metadata, parser, accessControl, queryExplainer, experimentalSyntaxEnabled, parameters);
+            Analysis analysis = analyzer.analyze(statement, true);
+
+            // get all parameters in query
+            List<Parameter> parameters = getParameters(statement);
+
+            // return the positions and types of all parameters.  If there are no parameters, return a single null row.
```

Comment:
The spec doesn't have very much detail, but generally for SQL when there are no results you return no rows.
This is what the specification says:
>To provide the necessary information to applications, SQL permits an application to request the SQL-server to describe a prepared statement. The description of a statement identifies the number of input dynamic parameters (describe input) and their data type information or it identifies the number of output dynamic parameters or values to be returned (describe output) and their data type information.